### PR TITLE
feat: add search and filtering on markets page

### PR DESCRIPTION
**Pull request by berteotti**
image

### CODE REVIEW

File: app/page.tsx [hunk 1]
```diff
@@ -8,37 +8,126 @@ import {
...
       getMarkets({
-        first: 10,
+        first: ITEMS_PER_PAGE,
         skip: 0,
-        orderBy: FixedProductMarketMaker_OrderBy.CreationTimestamp,
+        orderBy: selectedOption.orderBy,
         orderDirection: OrderDirection.Desc,
+        title_contains_nocase: search,
       }),
   });
```
Comment (ElRodrigote) [inline app/page.tsx:57]: WDYT about adding a debounce here (for 500ms or 700ms) to reduce the amount of calls we do to query data?

File: app/page.tsx [hunk 2]
```diff
@@ -8,37 +8,126 @@ import {
...
+          )}
         </div>
       </div>
+      {isLoading ? (
+        <div className="grid w-full grid-cols-1 gap-4 sm:grid-cols-2 xl:grid-cols-3 2xl:grid-cols-4">
+          {Array.from(Array(ITEMS_PER_PAGE).keys()).map((index) => (
+            <LoadingCardMarket key={index} />
+          ))}
+        </div>
+      ) : markets && markets.length > 0 ? (
```
Comment (ElRodrigote) [inline app/page.tsx:118]: Minor non blocking, this should be the same

```suggestion
      ) : markets?.length ? (
```

File: app/page.tsx [hunk 3]
```diff
@@ -8,37 +8,126 @@ import {
...
+                    <div
+                      key={option.name}
+                      onClick={() => setSelectedOption(option)}
+                      className="flex items-center justify-start px-3 py-2 space-x-2 font-semibold cursor-pointer"
+                    >
+                      <Icon
+                        className={
+                          selectedOption.name !== option.name
+                            ? "invisible"
+                            : undefined
```
Comment (ElRodrigote) [inline app/page.tsx:94]: Minor, non-blocking: 
This should behave the same
```suggestion
                        className={cx({
                          invisible: selectedOption.name !== option.name,
                        })}
```

File: app/page.tsx [hunk 4]
```diff
@@ -8,37 +8,126 @@ import {
...
+            <Button variant="pastel" className="space-x-2">
+              <p>{selectedOption.name}</p>
+              <Icon name="chevron-down" />
+            </Button>
+          )}
         </div>
       </div>
+      {isLoading ? (
+        <div className="grid w-full grid-cols-1 gap-4 sm:grid-cols-2 xl:grid-cols-3 2xl:grid-cols-4">
+          {Array.from(Array(ITEMS_PER_PAGE).keys()).map((index) => (
```
Comment (ElRodrigote) [inline app/page.tsx:114]: Minor, non-blocking:
This should behave the same, I think any of these options feel simpler and easier to read

```{[...Array(ITEMS_PER_PAGE)].map((index) => (```
```{Array(ITEMS_PER_PAGE).fill('').map((index) => (```
```{Array.from({ length: ITEMS_PER_PAGE}).map((index) => (```

**Review by ElRodrigote — APPROVED** [review]

**Comment by Diogomartf** [comment]
Looking good,

image

I would make this 2 column on mobile, I think it will work better.

**Comment by berteotti** [comment]
> I would make this 2 column on mobile, I think it will work better.

Made a different approach, preventing text from wrapping. I think it looks good this way, let me know what you think

**Review by Diogomartf — APPROVED** [review]

**Review by ElRodrigote — APPROVED** [review]

**Pull request merged**